lib/util.js

```diff
@@ -139,11 +140,21 @@ class ProgressLogger {
   }
 }
 
+function checkNodeCompatibility(_nodeVersion) {
+  const nodeVersion = _nodeVersion || process.versions.node;
+  if (!semver.gtr(nodeVersion, '8.*')) {
```

Comment:
Can we do both package.json's engine field as well?
Ideally, we read from that field, and validate here.

* the engines field is where we expect this
* NPX i believe doesn't respect the engines field

Reply:
We are checking the node version of the current environment where the `supported` tool is run right.
To check if the package we are testing is above 8, we have a node test that checks the latest node LTS anyways.
I can run the `supported` command from anywhere in the machine, may run it from the `ember-cli` folder to check if `broccoli` is in the support policy. Where the `engine` field of `ember-cli` is irrelevant for us. 

Let me know if I am missing something,

Reply:
the engine field is not an ember-cli thing. it's a core node/package.json feature
https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines

Reply:
I understand that. 

What I am asking which `engine` property are we reading?

Reply:
I think there is value in setting the `engine` in package.json to inform other developers of the node requirement.

Other than that, if NPX ignores the package.json `engine` field, is there any value in doing anything (reading or setting) of any `engine` field?

Reply:
yes.
if `A` adds `B` as dependency then `B` must meet `engine` requirements in `A`. This will still remain true in our case as we have set `supported's` package.json is pointing `>= 10.*`. 

This tool can be run without being added as a dependency, in that case, I feel it is redundant to check the `engine` and `process.versions.node`

Reply:
we should compared supported's `packagein.json#engine` with `process.node` and error if `process.version` is not within the supported's `package.json#engine` contraints

Reply:
Cool. That's true. Rather than comparing to a fixed node version, we can depend on this package's package.json#engine.